WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAddParameterViewModel.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/// View model for `BlazeAddParameterView`
+final class BlazeAddParameterViewModel: ObservableObject {
+    @Published var key: String
+    @Published var value: String
+    @Published var hasValidationError: Bool = false
+    @Published var hasCountError: Bool = false
+
+    let remainingCharacters: Int
+    let parameter: BlazeAdURLParameter?
+
+    var shouldDisableSaveButton: Bool {
+        key.isEmpty || value.isEmpty || hasCountError || hasValidationError
+    }
+
+    typealias BlazeAddParameterCompletionHandler = (_ key: String, _ value: String) -> Void
+    private let completionHandler: BlazeAddParameterCompletionHandler
+
+    private let cancellationHandler: () -> Void
+
+    init(remainingCharacters: Int,
+         parameter: BlazeAdURLParameter? = nil,
+         onCancel: @escaping () -> Void,
+         onCompletion: @escaping BlazeAddParameterCompletionHandler) {
+        self.remainingCharacters = remainingCharacters
+        self.parameter = parameter
+        self.cancellationHandler = onCancel
+        self.completionHandler = onCompletion
+
+        key = parameter?.key ?? ""
+        value = parameter?.value ?? ""
+    }
+
+    func didTapCancel() {
+        cancellationHandler()
+    }
+
+    func didTapSave() {
+        completionHandler(key, value)
+    }
+
+    func validateInputs() {
+        validateParameters()
+        validateInputLength()
+    }
+
+    /// This function validates the URL parameters using String.isValidURL().
+    /// As isValidURL() needs a full URL, we add Constant.baseURLForValidation is added.
+    private func validateParameters() {
+        let url = "\(Constant.baseURLForValidation)\(key)=\(value)"
+        hasValidationError = !url.isValidURL()
+    }
+
+    private func validateInputLength() {
+        let totalInputString = key + "=" + value
+        hasCountError = remainingCharacters - totalInputString.count <= 0
```

Comment:
If the user adds just enough characters, it should not be an error right?

```suggestion
        hasCountError = remainingCharacters - totalInputString.count < 0
```

Reply:
Correct, good point!